js/src/components/paid-ads/asset-group/asset-group-card.js

```diff
@@ -96,10 +96,24 @@ export default function AssetGroupCard() {
 		firstErrorRef.current = ref;
 	}
 
+	// Currently, because WC's `Form` component can not properly set the multiple values
+	// synchronously. So, it needs to distinguish whether the `baseAssetGroup` is already
+	// set when mounting and not being cleaned. If yes, it means the `values` are also
+	// available and include the changes made by the user. Otherwise, it should need to
+	// wait for the `baseAssetGroup` to be set from the `AssetGroupSection`. So that the
+	// edited values in `AssetField` won't be reset after remounting the `AssetGroupCard`.
+	const isSelectedAssetGroupInitiallyRef = useRef( isSelectedFinalUrl );
```

Comment:
💅  I found this explanations a bit confusing. It took me a while to understand the relation between `isSelectedFinalUrl` if `baseAssetGroup` is set or not.

On my understanding 

-  if `assetGroup` is set, means `values` are also available. And we use them as current values. 
- `assetGroup` is set when `isSelectedFinalUrl` is true when the component mounts.  
- We save this in a ref `isSelectedAssetGroupInitiallyRef ` for preventing it to be override in further component re-renders. 
- In case `assetGroup` is not initially set we load `baseAssetGroup` as current values